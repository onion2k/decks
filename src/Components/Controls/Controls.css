
.buttons {
    display: grid;
    grid-template-columns: repeat(9, 1fr);
    grid-gap: 15px;
}

.button {
    background-color: #000;
    color: #ddd;
    border: 1px solid #ddd;
    padding: 12px;
    font-size: 1.5em;
    text-align: center;
}

.play {
    grid-column: span 2;
}

.toggle {
    color: #666;
    background-color: #000;
    border: 1px solid #ddd;
    padding: 12px;
    font-size: 1.5em;
    text-align: center;
}

.toggle.on {
    color: #4f4;
}

.button:hover, .toggle:hover {
    background-color: #444;
    cursor: pointer;
}
