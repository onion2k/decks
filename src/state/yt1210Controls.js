import { configure, decorate, observable, action } from "mobx";

import fx from "./yt1210Sounds";

configure({ enforceActions: true });

export default class yt1210Controls {

    constructor(pm){
      this.pm = pm;
    }
    
    videoId = null;
  
    playing = false;
    repeat = false;
    shuffle = false;
    crackle = true;
    video = true;
  
    // play track
    play = (id = null) => {
      if (id!==null) {
        this.videoId = this.pm.getCurrentTrack().videoId;
      } else {
        this.videoId = this.pm.getTrackById(id).videoId;
      }
      this.playing = true;
    };
  
    // pause track
    pause = () => {
      this.playing = !this.playing;
    };
  
    // stop track
    stop = () => {
      this.playing = false;
    };
  
    // next track
    next = () => {
      this.videoId = this.pm.getNextTrack().videoId;
      this.playing = true;
    };
  
    // copy playlist from youtube

    toggle_repeat = () => {
      this.repeat = !this.repeat;
      if (this.repeat) { fx.ogg_scratchin.play(); }
    }
  
    toggle_shuffle = () => {
      this.shuffle = !this.shuffle;
    };
  
    toggle_crackle = () => {
      this.crackle = !this.crackle;
    };

    toggle_video = () => {
      this.video = !this.video;
    };
  }
  
  decorate(yt1210Controls, {
    play: action,
    pause: action,
    stop: action,
    next: action,
    toggle_repeat: action,
    toggle_shuffle: action,
    toggle_crackle: action,
    toggle_video: action,
    videoId: observable,
    playing: observable,
    crackle: observable,
    video: observable,
    repeat: observable,
    shuffle: observable
  });
  
