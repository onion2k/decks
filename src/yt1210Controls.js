// play
// stop
// pause
// next
// previous

// tonearm move
