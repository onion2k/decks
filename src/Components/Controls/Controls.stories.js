import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import { withKnobs, text, boolean, number } from "@storybook/addon-knobs";

import StoryRouter from "storybook-react-router";
import { linkTo } from "@storybook/addon-links";

import Controls from "./index.js";

const onPlayVideo = action("play video");
const onPauseVideo = action("pause video");
const onStopVideo = action("stop video");
const onChangeVideo = action("change video");
const onToggle = action("toggle");

storiesOf("Controls", module)
  .addDecorator(withKnobs)
  .addDecorator(
    StoryRouter(
      {
        "/playlists": linkTo("playlists", "Basic")
      },
      { initialEntries: ["/playlists"] }
    )
  )
  .add("about", () => {
    // const shuffle = boolean("Shuffle", false, "toggle");

    return (
      <Controls
        onStopVideo={onStopVideo}
        onPlayVideo={onPlayVideo}
        onPauseVideo={onPauseVideo}
        onChangeVideo={onChangeVideo}
        onToggle={onToggle}
        repeat={boolean("Repeat", false, "toggle")}
        shuffle={boolean("Shuffle", false, "toggle")}
        crackle={boolean("Crackle", false, "toggle")}
        video={boolean("Video", false, "toggle")}
      />
    );
  });
