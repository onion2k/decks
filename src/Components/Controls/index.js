import React, { Component } from 'react';
import { inject, observer } from 'mobx-react';
import { Link } from 'react-router-dom';
import './Controls.css';

class Controls extends Component {
    render() {

        let playlistsLink, playlistsIcon;

        if (window.location.pathname === '/playlists') {
            playlistsIcon = 'fa fa-music';
            playlistsLink = '/';   
        } else {
            playlistsIcon = 'fa fa-list';
            playlistsLink = '/playlists';
        }

        return (
        <div className="buttons">

            <Link to={ playlistsLink } className='button'><i className={ playlistsIcon } aria-hidden="true"></i></Link>

            <div className='button play' onClick={ ()=>{ this.props.onPlayVideo(true); }}><i className="fa fa-play" aria-hidden="true"></i></div>
            <div className='button' onClick={ ()=>{ this.props.onPauseVideo(); }}><i className="fa fa-pause" aria-hidden="true"></i></div>
            <div className='button' onClick={ ()=>{ this.props.onStopVideo(); }}><i className="fa fa-stop" aria-hidden="true"></i></div>
            <div className='button' onClick={ ()=>{ this.props.onChangeVideo(); }}><i className="fa fa-step-forward" aria-hidden="true"></i></div>

            <div className={'toggle'+(this.props.yt1210Store.repeat?' on':'')} onClick={ ()=>{ this.props.yt1210Store.repeat = !this.props.yt1210Store.repeat; }}><i className="fa fa-repeat" aria-hidden="true"></i></div>
            <div className={'toggle'+(this.props.yt1210Store.shuffle?' on':'')} onClick={ ()=>{ this.props.yt1210Store.shuffle = !this.props.yt1210Store.shuffle; }}><i className="fa fa-random" aria-hidden="true"></i></div>
            <div className={'toggle'+(this.props.yt1210Store.crackle?' on':'')} onClick={ ()=>{ this.props.yt1210Store.crackle = !this.props.yt1210Store.crackle; }}><i className="fa fa-headphones" aria-hidden="true"></i></div>
            <div className={'toggle'+(this.props.yt1210Store.video?' on':'')} onClick={ ()=>{ this.props.yt1210Store.video = !this.props.yt1210Store.video; }}><i className="fa fa-youtube" aria-hidden="true"></i></div>

        </div>
        );
    }
}

export default inject("yt1210Store")(observer(Controls));
