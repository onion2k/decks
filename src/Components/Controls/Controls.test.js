import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from 'react-router-dom'
import Enzyme, { mount } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() })

import Controls from ".";

import yt1210Controls from "../../state/yt1210Controls"
const controlState = new yt1210Controls();

describe("Playlist Controls", () => {
  const onClick = jest.fn();

  it("renders properly", () => {
    const controls = renderer
      .create(
        <MemoryRouter>
          <Controls playlistControls={controlState} />
        </MemoryRouter>
      )
      .toJSON();
    expect(controls).toMatchSnapshot();
  });

  it("should toggle options on click", () => {
    const controls = mount(
      <MemoryRouter>
        <Controls playlistControls={controlState} />
      </MemoryRouter>
    );

    expect(controls.find(".buttons > .button")).toHaveLength(9);

  });
});
