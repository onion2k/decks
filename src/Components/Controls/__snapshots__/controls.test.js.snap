// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist Controls renders properly 1`] = `
<div
  className="buttons"
>
  <a
    className="button"
    href="/playlists"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-list"
    />
  </a>
  <div
    className="button play"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-play"
    />
  </div>
  <div
    className="button"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-pause"
    />
  </div>
  <div
    className="button"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-stop"
    />
  </div>
  <div
    className="button"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-step-forward"
    />
  </div>
  <div
    className="button toggle"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-repeat"
    />
  </div>
  <div
    className="button toggle"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-random"
    />
  </div>
  <div
    className="button toggle on"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-headphones"
    />
  </div>
  <div
    className="button toggle on"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="fa fa-youtube"
    />
  </div>
</div>
`;
